### DISCUSSION TITLE
Xbox calibration data masked in Xidi-WinMM?

### DISCUSSION THREAD

**Discussion by rccraigb** [samuelgr/Xidi#56] [Q&A]
Using Windows 10's "Game Controllers" calibration function, I intentionally mis-calibrated several controllers in order to understand application behavior using Xidi-WinMM.  Interestingly, the bad calibration (sticks only allowed to travel left or forward, leaving right and backward fully saturated) was completely ignored by Xidi-WinMM.  That is, stick movement was 100% smooth, whereas tests without Xidi-WinMM reflected expected behavior - movements jumping wildly in the right and backward directions.

Removing the DLL and using the default OS library yielded testing results consistent with the bad calibration.  Does Xidi-WinMM mask or ignore calibration data?

**Comment by samuelgr** [comment]
I'm guessing the "bad calibration" is being applied to XInput controllers specifically?

For XInput controllers, Xidi gets controller data directly from XInput. I imagine XInput might ignore the calibration data, which is why Xidi-WinMM also appears to ignore it.

**Reply by rccraigb** [reply]
Yes, apologies for not specifying in the original question:  the devices are all various flavors of Xbox controllers (360, One, Razer), so yes XInput.

Thank you, by the way, for Xidi and your support of it:  very much appreciated!

**Comment by 80T** [comment]
Is it not a thing that XInput controllers are reporting 2 different sets of controls to the OS, like XI and DI independently.. ?
For example exposing two axes (one per each Trigger) in the XInput set, but one combined axis as both Triggers in the DInput set ?
Therefore, the axes calibration for the DInput controls set (which I assume is what we're seeing in the Gamepad CPL) is independent from XI ?
